strawberryfields/backends/tfbackend/ops.py

```diff
@@ -408,17 +364,10 @@ def fock_state(n, D, pure=True, batched=False):
     return fock
 
 
-def coherent_state(alpha, D, pure=True, batched=False):
+def coherent_state(r, phi, D, pure=True, batched=False):
     """creates a single mode input coherent state"""
-    coh = tf.stack(
-        [
-            tf.exp(-0.5 * tf.math.conj(alpha) * alpha)
-            * _numer_safe_power(alpha, n)
-            / tf.cast(np.sqrt(factorial(n)), def_type)
-            for n in range(D)
-        ],
-        axis=-1,
-    )
+    alpha = tf.cast(r, def_type) * tf.exp(1j * tf.cast(phi, def_type))
```

Comment:
(Minor suggestion): the casting of `r` and `phi` could be defined prior to this line